### PR TITLE
feat: make accessTokenPath configurable

### PR DESCRIPTION
**Pull request by Qdea**
## Summary of change

Due to implementation differences, when using the supertokens-node package, it would make sense to allow the user to configure the accessTokenPath instead of using the path statically. In the current PR, an optional accessTokenPath parameter has been added to TypeNormalisedInput, when specified, a custom path will be used in the user config. If the parameter is not specified, then the default value is taken, which was in the package before the changes.

In addition, minor syntax bugs have been fixed and the documentation in the docs folder, generated using npm run build-docs, has been updated.

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: lib/ts/recipe/session/recipe.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export default class SessionRecipe extends RecipeModule {
 
         this.isInServerlessEnv = isInServerlessEnv;
 
-        if (this.config.jwt.enable === true) {
+        if (this.config.jwt.enable) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/recipe.ts:77]: this is an unrelated change. Please revert it :)

File: lib/ts/recipe/session/types.ts [hunk 2]
```diff
@@ -129,6 +129,7 @@ export type TypeInput = {
 
 export type TypeNormalisedInput = {
     refreshTokenPath: NormalisedURLPath;
+    accessTokenPath?: NormalisedURLPath;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/types.ts:132]: the normalised input should not be optional. You should add `accessTokenPath` to the `TypeInput` type, and then in the `validateAndNormaliseUserInput` function [here](https://github.com/supertokens/supertokens-node/blob/1119aa4866/lib/ts/recipe/session/utils.ts#L127), you should set the normalised input to this value if provided by the user, else to `normaliseURLPathOrThrowError("/")`

File: lib/ts/recipe/session/utils.ts [hunk 3]
```diff
@@ -207,7 +207,7 @@ export function validateAndNormaliseUserInput(
     let accessTokenPayloadJWTPropertyName = "jwt";
     let issuer: string | undefined;
 
-    if (config !== undefined && config.jwt !== undefined && config.jwt.enable === true) {
+    if (config !== undefined && config.jwt !== undefined && config.jwt.enable) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/utils.ts:210]: please revert this change as it's not related to this PR

File: lib/ts/recipe/session/cookieAndHeaders.ts [hunk 4]
```diff
@@ -167,7 +165,7 @@ export function setCookie(
     if (pathType === "refreshTokenPath") {
         path = config.refreshTokenPath.getAsStringDangerous();
     } else if (pathType === "accessTokenPath") {
-        path = "/";
+        path = config.accessTokenPath ? config.accessTokenPath.getAsStringDangerous() : "/";
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/cookieAndHeaders.ts:168]:
```suggestion
        path = config.accessTokenPath.getAsStringDangerous() === "" ? "/" : config.accessTokenPath.getAsStringDangerous();
```

Comment (rishabhpoddar) [reply lib/ts/recipe/session/cookieAndHeaders.ts:168]: This change is suggested cause we will not make `accessTokenPath` optional in the normalised input type

File: lib/ts/recipe/session/cookieAndHeaders.ts [hunk 5]
```diff
@@ -143,14 +143,12 @@ export function setHeader(res: BaseResponse, name: string, value: string) {
 
 /**
  *
+ * @param config
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/cookieAndHeaders.ts:146]: please also consider adding a test about this new setting.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/session/cookieAndHeaders.ts (hunk 5 above)
Comment (Qdea) [reply lib/ts/recipe/session/cookieAndHeaders.ts:146]: Hi, this is not new setting, sorry. It just my IDE change this
image
bc of things like that
image
automatically.

File: lib/ts/recipe/session/utils.ts (hunk 3 above)
Comment (Qdea) [reply lib/ts/recipe/session/utils.ts:210]: Ok, will do

File: lib/ts/recipe/session/recipe.ts (hunk 1 above)
Comment (Qdea) [reply lib/ts/recipe/session/recipe.ts:77]: Ok, will do

File: lib/ts/recipe/session/cookieAndHeaders.ts (hunk 4 above)
Comment (Qdea) [reply lib/ts/recipe/session/cookieAndHeaders.ts]: Done

File: lib/ts/recipe/session/types.ts (hunk 2 above)
Comment (Qdea) [reply lib/ts/recipe/session/types.ts]: Done

File: lib/ts/normalisedURLPath.ts [hunk 6]
```diff
@@ -43,7 +43,7 @@ export default class NormalisedURLPath {
     };
 }
 
-function normaliseURLPathOrThrowError(input: string): string {
+export function normaliseURLPathOrThrowError(input: string): string {
```
Comment (rishabhpoddar) [inline lib/ts/normalisedURLPath.ts:46]: You do not need to export this since the `NormalisedURLPath` class is already exported. Use that instead

File: lib/ts/recipe/session/utils.ts [hunk 7]
```diff
@@ -229,6 +232,7 @@ export function validateAndNormaliseUserInput(
 
     return {
         refreshTokenPath: appInfo.apiBasePath.appendPath(new NormalisedURLPath(REFRESH_API_PATH)),
+        accessTokenPath: appInfo.apiBasePath.appendPath(new NormalisedURLPath(accessTokenPath)),
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/utils.ts:235]:
```suggestion
        accessTokenPath,
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/session/cookieAndHeaders.ts (hunk 5 above)
Comment (rishabhpoddar) [reply lib/ts/recipe/session/cookieAndHeaders.ts]: Ah no. I meant that you should add a test about testing the new accessTokenPath config

File: lib/ts/recipe/session/utils.ts (hunk 7 above)
Comment (Qdea) [reply lib/ts/recipe/session/utils.ts:235]: So i dont need to append this?

File: lib/ts/recipe/session/cookieAndHeaders.ts (hunk 5 above)
Comment (Qdea) [reply lib/ts/recipe/session/cookieAndHeaders.ts]: and where should I put this?

Comment (rishabhpoddar) [reply lib/ts/recipe/session/cookieAndHeaders.ts]: Here are a few examples: https://github.com/supertokens/supertokens-node/blob/master/test/config.test.js

Comment (Qdea) [reply lib/ts/recipe/session/cookieAndHeaders.ts]: Thanks, pushed tests

**Review by rishabhpoddar — APPROVED** [review]
